CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+# Contributing to the documentation
+
+## Repository structure
+
+Briefly, this is what you need to know about the documentation repository structure before contributing:
+
+* The documentation is module-based, which means most sections are individual modules. These modules are available in the `/docs/topics/modules/` folder.
+* The modules are linked together to form a chapter, document, or guide using assemblies. These assemblies are asciidoc files in the `/docs/topics/` folder. For example, the `/docs/topics/getting-started-guide.adoc` file is the main assembly for the Getting Started Guide and links to other, smaller assemblies for parts or chapters within the guide.
+* Module files naming conventions closely follow the title of the corresponding section. For example, a section in the Getting Started document with the title *Working with pipelines* is named **working_with_pipelines.adoc** in the *modules* folder. In some cases, this is not a direct match, so to confirm the name of the asciidoc file for a section, click the section title and the ID after the *#* in the URL is the name of the module file. For example, a URL such as https://docs.openshift.io/getting-started-guide.html#viewing_build_pipeline_oso has a slightly different title but the ID after the hash symbol (**viewing_build_pipeline_oso**) is the name of the file.
+
+## Conventions and style guidelines
+
+To get started quickly, we recommend reviewing the following conventions that are relevant for this documentation suite:
+
+**Writing conventions**
+
+* For instructions, state the location, then action and subject. For example:
+  * Incorrect: *Click OK in the XYZ tab to continue.*
+  * Correct: *In the XYZ tab, click OK to continue.*
+* Use gerunds in titles when possible. For example:
+  * Incorrect: *1.1 Create a new project*
+  * Correct: *1.1 Creating a new project*
+* All titles should be in sentence case. Exceptions for words that are proper nouns. For example:
+  * Incorrect: *1.2 Configuring Your Che Workspace*
+  * Correct: *1.2 Configuring your Che workspace*
+* URLs must be bared in the text, not hidden. For example:
+  * Incorrect: *Navigate to the [Red Hat website](www.redhat.com).* 
+  * Correct: *Navigate to (www.redhat.com).*
```

Comment:
Ideally would be great to use a table here, but for now this is just fine.

Reply:
I'd switch to a table if there were significantly more entries. At the moment, since it's just four or so entries, it seems useful to just have a list and keep it brief.

Reply:
Sure.

Reply:
@mishaone Where does the rule about bare URLs come from? I believe the convention is to hide the URL behind a meaningful link text (correct: "Red Hat website", incorrect: "here"). A bare URL should only be used when there is a reason, such as "type 'https://redhat.com' in the address bar".

Reply:
+1 on not using a table. Tables should be avoided when reasonably possible. They are hard to translate, scale badly, and their source code is hard to navigate.

Reply:
@rkratky the bared URLs was from a mailing list discussion a long time ago. I'm happy to discard it if it isn't a rule anymore. From memory, I think it was because PDFs and other format printed out effectively "hid" the URLs if we didn't expose them.

Reply:
@mishaone PDF output should be smart enough to use bare URL regardless of the source (or at least offer clickable links -- which ours does). For web output, I believe we should be as user friendly as possible.